### DISCUSSION TITLE
No configuration file??

### DISCUSSION THREAD

**Discussion by BarbzYHOOL** [ccache/ccache#263] [Support]
debian 8 jessie
ccache version 3.1.12
clang 3.5

I would like to increase cache size etc but when I type "ccache -s", there is no conf file (one friend showed me his stats and there were extra lines for conf file):

```
cache directory                     /root/.ccache
cache hit (direct)                   113
cache hit (preprocessed)               3
cache miss                           566
called for link                       41
compile failed                        25
preprocessor error                     1
files in cache                         0
cache size                             0 Kbytes
max cache size                       1.0 Gbytes

```

I tried to create conf files in:

> The cache-specific configuration file <ccachedir>/ccache.conf (typically $HOME/.ccache/ccache.conf).
> The system-wide configuration file <sysconfdir>/ccache.conf (typically /etc/ccache.conf or /usr/local/etc/ccache.conf).

But it's not read or I don't know how to reload it.

What can I do?

PS: I want to make it with configuration files so I can easily deploy it with ansible

**Comment by jrosdahl** [comment]
The first ccache version with config file support is 3.2 (see the [release notes for ccache 3.2](https://ccache.samba.org/releasenotes.html#_ccache_3_2)), so you need to upgrade ccache if you want to use configuration files.

**Comment by BarbzYHOOL** [comment]
Ok thank you, will do!

**Comment by BarbzYHOOL** [comment]
Hmm, what is the process to update Ccache on debian 8? the stable version in repo is 3.1.12

**Comment by jrosdahl** [comment]
[Build and install it from source code yourself](https://github.com/ccache/ccache/blob/master/doc/INSTALL-from-release-archive.md) or backport a deb package from Debian 9, I guess.

**Comment by BarbzYHOOL** [comment]
Can I compile from source even if I use an old Clang? like 3.5.0

**Comment by jrosdahl** [comment]
I would be surprised if you couldn't.

**Comment by BarbzYHOOL** [comment]
Sorry, I'm really unprecise, I meant, will the new ccache work with an old Clang?

**Comment by jrosdahl** [comment]
Newer ccache versions have much better support for Clang, so probably yes.

**Comment by BarbzYHOOL** [comment]
Ok nice.

So new problem now, the documentation says:

> You may set the installation directory and other parameters by options to ./configure. To see them, run ./configure --help.

But:

    bash: ./configure: No such file or directory

I'm in the ccache dir and there are:

```
-rw-r--r-- 1 root root  44K mai   20 13:31 config.guess
-rw-r--r-- 1 root root  36K mai   20 13:31 config.sub
-rw-r--r-- 1 root root 4,8K mai   20 13:31 configure.ac

```

**Comment by jrosdahl** [comment]
Looks like you're trying to build from the Git repo, but the instructions I linked to are for building from a release archive found here: https://ccache.samba.org/download.html

**Comment by BarbzYHOOL** [comment]
lol ok

So when I make (the 3.4 version) I got a lot of warnings:

```
11 warnings generated.
ccache /usr/bin/clang -DHAVE_CONFIG_H -DSYSCONFDIR=/usr/local/etc -I. -I./src -I./unittest  -g -O2 -Wall -W -c -o src/version.o src/version.c
clang: warning: argument unused during compilation: '-I .'
clang: warning: argument unused during compilation: '-I ./src'
clang: warning: argument unused during compilation: '-I ./unittest'
ccache /usr/bin/clang -DHAVE_CONFIG_H -DSYSCONFDIR=/usr/local/etc -I. -I./src -I./unittest  -g -O2 -Wall -W -c -o src/getopt_long.o src/getopt_long.c
clang: warning: argument unused during compilation: '-I .'
clang: warning: argument unused during compilation: '-I ./src'
clang: warning: argument unused during compilation: '-I ./unittest'
ccache /usr/bin/clang -DHAVE_CONFIG_H -DSYSCONFDIR=/usr/local/etc -I. -I./src -I./unittest  -g -O2 -Wall -W -c -o src/hashtable.o src/hashtable.c
clang: warning: argument unused during compilation: '-I .'
clang: warning: argument unused during compilation: '-I ./src'
clang: warning: argument unused during compilation: '-I ./unittest'
ccache /usr/bin/clang -DHAVE_CONFIG_H -DSYSCONFDIR=/usr/local/etc -I. -I./src -I./unittest  -g -O2 -Wall -W -c -o src/hashtable_itr.o src/hashtable_itr.c
clang: warning: argument unused during compilation: '-I .'
clang: warning: argument unused during compilation: '-I ./src'
clang: warning: argument unused during compilation: '-I ./unittest'
ccache /usr/bin/clang -DHAVE_CONFIG_H -DSYSCONFDIR=/usr/local/etc -I. -I./src -I./unittest  -g -O2 -Wall -W -c -o src/murmurhashneutral2.o src/murmurhashneutral2.c
clang: warning: argument unused during compilation: '-I .'
clang: warning: argument unused during compilation: '-I ./src'
clang: warning: argument unused during compilation: '-I ./unittest'
ccache /usr/bin/clang -DHAVE_CONFIG_H -DSYSCONFDIR=/usr/local/etc -I. -I./src -I./unittest  -g -O2 -Wall -W -c -o src/snprintf.o src/snprintf.c
clang: warning: argument unused during compilation: '-I .'
clang: warning: argument unused during compilation: '-I ./src'
clang: warning: argument unused during compilation: '-I ./unittest'
ccache /usr/bin/clang -g -O2 -Wall -W -o ccache src/main.o src/args.o src/ccache.o src/cleanup.o src/compopt.o src/conf.o src/counters.o src/execute.o src/exitfn.o src/hash.o src/hashutil.o src/language.o src/lockfile.o src/manifest.o src/mdfour.o src/stats.o src/unify.o src/util.o src/version.o src/getopt_long.o src/hashtable.o src/hashtable_itr.o src/murmurhashneutral2.o src/snprintf.o   -lm  -lz

```
And more I couldn't paste

Should I still make install?

**Comment by jrosdahl** [comment]
Yes.

(Those warnings are because the old ccache version doesn't really support clang. The new version should do better.)

**Comment by BarbzYHOOL** [comment]
Ok thank you, I did it.

I still have another question:

when I installed it, "ccache --version" prints 3.4 but if I go on my other byobu tab (persistent session with tmux), it still says 3.1. What should I do to update this (besides closing my session)? it is probably only cosmetic but not sure. Just in case you know what to do since it's unrelated to ccache

**Comment by BarbzYHOOL** [comment]
I also suggest that you put a download link on the last release page called "Lastest stable release" and so we can wget it and get the lastest file automatically

**Comment by jrosdahl** [comment]
> What should I do to update this (besides closing my session)? it is probably only cosmetic but not sure. Just in case you know what to do since it's unrelated to ccache

Maybe `hash -r` so that your shell finds ccache in `/usr/local/bin`?

> I also suggest that you put a download link on the last release page called "Lastest stable release" and so we can wget it and get the lastest file automatically

I don't think that's a good idea, sorry.

**Comment by BarbzYHOOL** [comment]
Thank you for `hash -r`, I didn't know that command at all and it worked fine! I hope it didn't create some issues that will require me to relaunch my byobu session after 1hour of frustration though :P

May I ask you why it's not a good idea? just being curious, it's not a big deal for me anyway (since I'll probably install with .deb packages on newer debians)
